pkg/dogstatsd/replay/file.go

```diff
@@ -0,0 +1,52 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2021 Datadog, Inc.
+
+package replay
+
+import (
+	"bufio"
+	"fmt"
+
+	"github.com/h2non/filetype"
+)
+
+var (
+	datadogType = filetype.NewType("dog", "datadog/capture")
+	// DATADOG0F1FF0000 in HEX (D474D060F1FF0000); F1 for different versions, 00 to terminate header
+	datadogHeader = []byte{0xD4, 0x74, 0xD0, 0x60, 0xF1, 0xFF, 0x00, 0x00}
+)
+
+func init() {
+	// Register the new matcher and its type
+	filetype.AddMatcher(datadogType, datadogMatcher)
+}
+
+func datadogMatcher(buf []byte) bool {
+	if len(buf) < len(datadogHeader) {
+		return false
+	}
+
+	for i := 0; i < len(datadogHeader); i++ {
+		if buf[i] != datadogHeader[i] {
+			return false
+		}
+	}
+
+	return true
+}
+
+// WriteHeader writes the datadog header to the Writer argument to conform to the .dog file format.
+func WriteHeader(w *bufio.Writer) error {
+
+	//Write header
+	if n, err := w.Write(datadogHeader); err != nil || n < len(datadogHeader) {
```

Comment:
The condition `n < len(datadogHeader)`  is not required as `WriteHeader` returns nil when `err != nil`.